File: Hackerrank/Ice Cream Parlor/Ice Cream Parlor.c
//NOTE: took my own i/o

#include<stdio.h>
int main()
{
    int t,c,l,i,j,arr[20000];
    for(scanf("%d",&t);t>0;t--) //take input
    {
        scanf("%d%d",&c,&l);
        for(i=0;i<l;i++)  //iterate through array to take array value inputs
        scanf("%d",&arr[i]);
        for(i=0;i<l-1;i++) //iterate through array inner and outer, O(n^2)
        for(j=i+1;j<l;j++)
        {
            if(arr[i]+arr[j]==c)   //if sum found i.e sum of two ice creams is the money they have
            printf("%d %d\n",i+1,j+1);
        }
    }
    return 0;
}

File: Hackerrank/Ice Cream Parlor/Ice Cream Parlor.py
"""
Two friends like to pool their money and go to the ice cream parlor. They always choose two distinct flavors and they spend all of their money.
Given a list of prices for the flavors of ice cream, select the two that will cost all of the money they have.
"""

def icecreamParlor(m, arr):
    # Create an empty dictionary
    s={}
    for i in range (len(arr)):
        # m-arr[i] would complement arr[i] to make sum of money m
        if m-arr[i] in s:
            # required Flavors infex ( 1 based index)
            return [s[m-arr[i]],i+1]
        else:
            # else: add it to dictionary s: (here 1 is added becz it is 1 based indexing )
            s[arr[i]]=i+1